pkg/process/procutil/process_windows_toolhelp.go

```diff
@@ -263,3 +263,21 @@ func (cp *cachedProcess) close() {
 		cp.procHandle = windows.Handle(0)
 	}
 }
+
+// GetParentPid looks up the parent process given a pid
+func GetParentPid(pid uint32) (uint32, error) {
+	var pe32 w32.PROCESSENTRY32
+	pe32.DwSize = uint32(unsafe.Sizeof(pe32))
+
+	allProcsSnap := w32.CreateToolhelp32Snapshot(w32.TH32CS_SNAPPROCESS, 0)
```

Comment:
should we be locking to the current OS thread for subsequent accesses to this handle?